auction-app-frontend/src/pages/Shop/Shop.jsx

```diff
@@ -1,5 +1,159 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate, useLocation } from "react-router-dom";
+
+import { 
+  Button, 
+  Checkbox, 
+  ProductGrid,
+  ErrorComponent,
+  LoadingComponent 
+} from "src/components";
+
+import {
+  getProductsPaginated,
+  getCategoriesWithSubcategories
+} from "src/services";
+import { collapse, expand } from "src/assets/icons";
+
+import "./style.scss";
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
+
 const Shop = () => {
-  return <div>Shop Page</div>;
+  const [activeCategory, setActiveCategory] = useState();
+  const [items, setItems] = useState([]);
+  const [page, setPage] = useState(0);
+  const [categories, setCategories] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+  const [productsError, setProductsError] = useState(null);
+  const [productsLoading, setProductsLoading] = useState(true);
+  const [categoriesError, setCategoriesError] = useState(null);
+  const [categoriesLoading, setCategoriesLoading] = useState(true);
+  const [checked, setChecked] = useState({});
+
+  const navigate = useNavigate();
+  const query = useQuery();
+
+  const categoryId = query.get("category");
+  const searchProduct = query.get("search_product");
+
+  const fetchProducts = () => {
+    setProductsLoading(true);
+    
+    getProductsPaginated(page, 9, categoryId, searchProduct)
+      .then((res) => {
```

Comment:
Since we have some time, let's do this. 
First avoid generic responses, if this returns products, rename `res` to `products` (apply this for all requests)
Also for errors, avoid `err` and use full name `error`